pyanaconda/installation.py

```diff
@@ -113,15 +114,20 @@ def doConfiguration(storage, payload, ksdata):
 
     # creating users and groups requires some pre-configuration.
     user_config = TaskQueue("User creation", N_("Creating users"))
-    user_config.append(Task("Configure root", ksdata.rootpw.execute, (storage, ksdata)))
-    user_config.append(Task("Configure user groups", ksdata.group.execute, (storage, ksdata)))
-    user_config.append(Task("Configure user", ksdata.user.execute, (storage, ksdata)))
-    user_config.append(Task("Configure SSH key", ksdata.sshkey.execute, (storage, ksdata)))
+
+    users_proxy = USERS.get_proxy()
+    users_dbus_tasks = users_proxy.InstallWithTasks(util.getSysroot())
+    # add one Task instance per DBUS task
+    for dbus_task in users_dbus_tasks:
+        task_proxy = USERS.get_proxy(dbus_task)
+        user_config.append(Task(task_proxy.Name, sync_run_task(task_proxy)))
```

Comment:
Aren't you running the task in the time of appending it to the queue while it should be run when the queue starts, ie
Task(task_proxy.Name, sync_run_task, (task_proxy,)) ?

Reply:
Indeed, it looks that way! And likely due to the two phases of the installation process, it even runs +- at the right time! (after partitioning & package installation).

Reply:
Fixed.